Collected small fixes and changes

akohlmey opened the pull request:
**Summary**

This pull request collects multiple small changes and fixes

**Related Issue(s)**

Fixes #4487
Fixes #4491 
Fixes #4489 
Fixes #4499 
Fix halt universe flag was inspired by this discussion https://matsci.org/t/quitting-all-partitions/61686/
SPIN package fixes and example update inspired by https://matsci.org/t/spin-example-input-script-crashing-at-time-step-0/61762

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- fix issue when trimming neighbor lists where the trimmed list request does not have a custom cutoff
- fix normalization issue with compute rdf when atom types were swapped on input
- fix bug when restarting from a general triclinic geometry
- do not allow using fix pimd/langevin in pimd mode with multiple processors per bead
- add test to ReaxFF parameter file parser to error out with a meaningful error when the "lgvdw" flag is used with an incompatible force field file
- add "universe" option to fix halt that will synchronize a stop across multiple partitions
- add note about limited dynamic group support for fix reaxff/species doc page
- set stores_ids flag to 1 for several fixes that store atom ID related or dependent data
- make certain nstyle member variable is initialized before use in RegionPlane class
- bugfix from @ndtrung81 to avoid problems when using read_dump in cases where not all MPI ranks receive atoms
- fix some minor memory leaks in the SPIN package
- correctly tally "Sync" time when enabled and thus don't double count it in "Other"
- enable recently release sphinx 8.2.x
- document (some) internal styles
- correct and clarify the documentation for bond styles harmonic/shift and harmonic/shift/cut
- documentation fixes and cleanups
- modernize some error messages
- update SPIN/read_restart examples and logfiles

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

akohlmey committed:
add test to detect if a potential file was incorrectly used with lgvdw yes

akohlmey committed:
add note that dynamic groups are only supported without averaging

akohlmey committed:
bugfix from Trung to avoid problems with read_dump when not all MPI ranks have atoms

akohlmey committed:
Merge remote-tracking branch 'github/develop' into collected-small-fixes

akohlmey committed:
make certain that nstyle is always initialized and detect partial variable setups

akohlmey committed:
make certain variable indices are initialized

akohlmey committed:
move documentation of sign() function to logical location and add version tag

akohlmey committed:
Merge branch 'develop' into collected-small-fixes

akohlmey committed:
enable use of Sphinx 8.2.x

akohlmey committed:
document some internal styles

akohlmey committed:
small correction

akohlmey committed:
Merge remote-tracking branch 'github/develop' into collected-small-fixes

akohlmey committed:
only trim if neighbor list request has a custom cutoff

akohlmey committed:
use pairwise cutoff for trimming unless a custom cutoff if given

akohlmey commented:
@stanmoore1 @jtclemm Please check out the changes to neighbor list trimming based on the bug report in #4487. In the test case the custom cutoff is not set for the neighbor list that is trimmed and then it is trimmed to cutoff 0.0.

I could not come up with an idea for how to do the same for "size" style neighbor lists, so I effectively disabled trimming when there is no custom cutoff.

stanmoore1 commented:
Neighlist changes likely need to be ported to KOKKOS

akohlmey committed:
use different method to enforce the C locale in LAMMPS GUI

akohlmey committed:
update statements about what packages support and what not

akohlmey commented:
> Neighlist changes likely need to be ported to KOKKOS

Yes, they do. I had a look, but it is beyond my level of understanding of KOKKOS to add access to data like the pairwise cutoff or atom type, so your assistance is needed here.

jtclemm commented:
> @stanmoore1 @jtclemm Please check out the changes to neighbor list trimming based on the bug report in #4487. In the test case the custom cutoff is not set for the neighbor list that is trimmed and then it is trimmed to cutoff 0.0.

Those changes look correct to me. I can't see any issues that could emerge.

> I could not come up with an idea for how to do the same for "size" style neighbor lists, so I effectively disabled trimming when there is no custom cutoff.

Can the same strategy, defining `cut_trim` as `cutneighsq[itype][jtype]` if no custom cutoff is defined, not work?

akohlmey committed:
reverse normalization between type pairs if the types were swapped on input

akohlmey committed:
modernize error messages

akohlmey commented:
> Can the same strategy, defining `cut_trim` as `cutneighsq[itype][jtype]` if no custom cutoff is defined, not work?

On second thought, I don't think it is necessary. In the "size" style neighborlist build, the cutoff to compare to is computed from `radius[i] + radius[j] + skin`, so there should be nothing to trim unless a shorter(!) custom cutoff is set. It would make no sense to look for a per-type cutoff.

akohlmey committed:
rewrite incorrect statement

akohlmey committed:
add "condense" keyword to delete_atoms what will compact the atom IDs while preserving their order

akohlmey committed:
error out when trying unsupported pimd method with multiple processors per bead

akohlmey committed:
add some more tests for delete_atom

akohlmey committed:
add option to synchronize fix halt stop requests across multi-partition runs

akohlmey committed:
modernize error messages

akohlmey committed:
add MPI stub for MPI_Iprobe()

akohlmey committed:
fix typo

akohlmey committed:
Merge remote-tracking branch 'github/develop' into collected-small-fixes

akohlmey committed:
set stores_ids flag for some fixes as suggested by @sjplimp

akohlmey committed:
restore original delete_atoms implementation. new version spun off as PR #4492

akohlmey committed:
Revert "add some more tests for delete_atom"

This reverts commit 2ff078f9730cf092e0c88580b707555182559e64.

akohlmey committed:
Merge branch 'develop' into collected-small-fixes

akohlmey committed:
fix broken URL due to typo

akohlmey committed:
document removed compute atom/molecule and its equivalent using chunks

akohlmey committed:
fix typos

akohlmey committed:
resolve spelling and formatting issues

akohlmey committed:
correct and clarify statements about the potentials and their relations to each other

akohlmey committed:
report which variable name was incorrect

akohlmey committed:
formatting

akohlmey committed:
properly tally the SYNC time contributions to the ALL time entry

this way its contribution is not double counted when determining
the "Other" time data.

akohlmey committed:
include space in custom formats to show how spaces can be used to separate columns

akohlmey committed:
fix minor memory leaks

akohlmey committed:
rework SPIN restart/data file examples

akohlmey committed:
improve error messages for easier debugging

akohlmey committed:
update examples for consistency

akohlmey committed:
add citation reminder to fix qtpie/reaxff

akohlmey committed:
modernize LaTeX for formatting text in math mode

akohlmey committed:
revert unintended commit

akohlmey committed:
Merge branch 'develop' into collected-small-fixes

akohlmey committed:
update .gitignore

akohlmey committed:
improve error messages

akohlmey committed:
must use delete[] on array allocated with new

akohlmey committed:
print detailed version string also for multi-replica runs

akohlmey committed:
update pimd/*/bosonic examples

akohlmey committed:
limit to 1 processor per partition

akohlmey committed:
free communicator only if not NULL

akohlmey committed:
apply clang-format

akohlmey committed:
modernize boldface font selection in LaTeX sections

akohlmey committed:
modernize use of \it and \sf

akohlmey committed:
small changes to appease coverity scan

akohlmey committed:
key word -> keyword

akohlmey committed:
clarify. mention using timer command to re-enable regular runs

akohlmey committed:
corrections for documentation of bosonic PIMD fix styles

akohlmey committed:
small clarification. add version tag for universe keyword

akohlmey committed:
fix bug missing a read_int() when reading general triclinic rotation matrix

akohlmey committed:
ensure that df[] array is always initialized
